src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.tsx

```diff
@@ -243,7 +248,7 @@ function ContactMethodDetailsPage({route}: ContactMethodDetailsPageProps) {
                     <MenuItem
                         title={translate('contacts.setAsDefault')}
                         icon={Star}
-                        onPress={setAsDefault}
+                        onPress={() => (isAccountLocked ? setIsLockedAccountModalOpen(true) : setAsDefault())}
```

Comment:
We should be careful to memoize anything that was a stable reference before (use `useCallback` here)